guides/dsl.md

```diff
@@ -0,0 +1,314 @@
+
+```javascript
+program.bind("sort the students by last name, then first name per teacher", ({find, record, gather}) => {
+  let student = find("student")
+  let ix = gather(student.firstName, student.lastName).per(student.teacher).sort("up", "down")
+  return [
+    record("html/element", {tagname: "div", sort: ix, text: `${student.firstName} ${student.lastName}`})
+  ];
+});
+```
+
+## Importing and Exporting Records 
+
+### Getting Data Into Eve - `inputEAVs()`
+
+You can import raw EAVs into Eve with the `inputEAVs()` function. Currently, `inputEAVs()` mut be called at least once to initialize your program, and it can only be used after you define all blocks. These limitations will be lifted in future versions.
+
+`inputEAVs()` takes as input a list of entity, attribute, value triples. The entity value identifies the record to which the attribute and value belong, so it must be unique to that record. For example:
+
+```javascript
+program.inputEavs([[0,"tag","person"],[0,"name","Archibald"])
+```
+
+This will create a record tagged "person" with the name attribute "Archibald".
+
+The `appendAsEAVs()` function allows you to destructure an object into a uniquely identified set of EAVs, which can then be input into Eve:
+
+```javascript
+import {appendAsEAVs} from "witheve";
+let inputs = [];
+let archibald = {tag: "person", name: "Archibald"}
+appendAsEAVS(inputs, archibald);
+program.inputEAVs(inputs);
+```
+
+### Getting Records out of Eve - Watchers
+
+Watchers are a third type of block available in the DSL. These allow you to monitor changes in specific records and react to them with a callback function:
+
+```javascript
+program.watch("Export information about students", ({find, lookup, record}) => {
+  // search for records tagged student
+  let student = find("student");
+  // lookup attributes and values related to each student
+  let {attribute, value} = lookup(student);
+  return [
+    // Add these attributes and values to the student, creating a diff to which we can react
+    student.add(attribute, value)
+  ];
+})
+// React to each addition or removal
+.asDiffs((diff) => {
+  for(let [e, a, v] of diff.adds) {
+    // ... do something ...
+  }
+  for(let [e, a, v] of diff.removes) {
+    // ... do something ...
+  }
+});
+```
+
+If you care about specific attributes, if may be more convenient to write a watcher with `asObjects()` instead of `asDiffs()`:
+
+```javascript
+  program.watch("Export student GPA", ({find, lookup, record}) => {
+    // search for records tagged student
+    let student = find("student" {name, GPA});
+    return [
+      // Add these attributes and values to the student, which creates a diff to which we can react
+      record("grade", {name, GPA})
+    ];
+  })
+  // Handle adds and removes as objects
+  .asObjects(({adds, removes}) => {
+    for(let key in adds) {
```

Comment:
I'd rename key to id for clarity -- this key will be the id of the exported record. In many cases you don't care, but in some that's useful to know.